subprojects/core/src/main/java/org/gradle/process/internal/DefaultJavaForkOptions.java

```diff
@@ -207,6 +207,15 @@ public void setDebug(boolean enabled) {
         options.setDebug(enabled);
     }
 
+    public int getPort() {
```

Comment:
❌ This API seems inconsistent - There's a `setDebugOptions` (which only takes a port, no other options) and then there's a `getPort` ("what port" I may ask as a user.

I think instead we should have a `getDebugOptions` here and then have `port`, `server` and `suspend` properties on that.

Reply:
`server` and `suspend` is the same thing right? `server=y` means the JVM is suspended until the debugger is attached. Nonetheless, I think we should not add the `suspend` property until it's hard to use (i.e. there's no progress event implemented yet).

We can still create a DebugOptions interface with a single `getPort()` method on it.

Reply:
Done.